pkg/runtime/runtime.go

```diff
@@ -1275,11 +1267,11 @@ func (a *DaprRuntime) processOneComponent(comp components_v1alpha1.Component) er
 	dependency := componentDependency(compCategory, comp.Name)
 	if deps, ok := a.pendingComponentDependents[dependency]; ok {
 		delete(a.pendingComponentDependents, dependency)
-		go func(dependents []components_v1alpha1.Component) {
-			for _, dependent := range dependents {
-				a.pendingComponents <- dependent
+		for _, dependent := range deps {
+			if err := a.processOneComponent(dependent); err != nil {
+				return err
 			}
-		}(deps)
+		}
```

Comment:
How about return the deps,  and handle them in the upper level method? I think it will be clearer

Reply:
True, this is changing the meaning of "processOneComponent", isn't it?

Reply:
So, in order to do this the caller will have to push the return deps into a fifo and keep iterating until the fifo is empty. It actually ends up being somewhat ugly on the caller and will require substantial changes to our unit tests as well. I think I'd rather just rename processOneComponent to processComponentAndDependants and keep it recursive.

Reply:
Yeah, it needs a FIFO and iteration.
I would suggest using iteration instead of recursion, which can also make the processing for one component completely and one by one.

But it's up to you.
My preferred option maybe look like following:
```
for comp := range a.pendingComponents {
		if comp.Name == "" {
			continue
		}
                todo := []components_v1alpha1.Component{comp}
                for len (todo) > 0 {
                                next, todo := todo[0], todo[1:]
                                deps, err :=a.processComponentAndDependents(next)
                                if err != nil { }
                                todo = append (todo, deps...)
                }
	}
```

Reply:
cc @youngbupark , what do you think of it.

Reply:
I went down this road, but stopped when I realized that I'd have to change all our unit tests which call processOneComponent to either follow this pattern, or expect to get a dependency list.

Reply:
> cc @youngbupark , what do you think of it.

I missed your ask, but I would say that if recursion would make the code simpler without deep depth, then I would keep it as is.